core/src/main/java/com/google/instrumentation/common/Timestamp.java

```diff
@@ -90,6 +96,23 @@ public int getNanos() {
     return nanos;
   }
 
+  /**
+   * Returns the {@code Timestamp} calculated as this {@code Timestamp} plus {@code nanos}.
+   *
+   * @param nanos the nanoseconds to be added to the current timestamp.
+   * @return the {@code Timestamp} calculated as this {@code Timestamp} plus {@code nanos}.
+   */
+  public Timestamp addNanos(long nanos) {
+    long newNanos = nanos + this.nanos;
```

Comment:
Do you want to worry about overflow here? (the case where nanos = Long.MAX_VALUE)